### ISSUE TITLE
Flow type incorrectly determined to be undefined

### ISSUE THREAD

**Issue by leebyron** [babel/babel-eslint#66]
Running eslint with babel-eslint and using flow types:

```js
type SomeNewType = any;
```

will fail the `no-undef` lint with `'SomeNewType' is not defined`

Appears to be because `TypeAlias` AST node walks over the `id` field which is `Identifier` and eslint mistakes it for an undefined variable usage.

Not sure how exactly to proceed with a fix for this - should babel process out the flow annotations from the AST before passing to eslint?

**Comment by sebmck** [comment]
Closed via #67.
